src/lex.cpp

```diff
@@ -628,18 +632,22 @@ void lexer::parse_binary_number() {
     input += 1;
   }
 
+  report_garbage_in_number_literal(input);
+}
+
+void lexer::report_garbage_in_number_literal(const char8* input) {
```

Comment:
Nit: I think `check_garbage_in_number_literal` is a better name, since this function doesn't always report an error.

Reply:
Updated in latest commit. I agree that is a better name.